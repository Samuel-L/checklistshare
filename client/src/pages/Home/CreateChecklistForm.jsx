import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from 'material-ui/styles';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';

import ChecklistItem from './ChecklistItem';

const styles = theme => ({
  textField: {
    margin: '5px',
  },
  buttonGroup: {
    [theme.breakpoints.up('md')]: {
      width: '150px',
      float: 'right',
    },
    [theme.breakpoints.down('sm')]: {
      position: 'absolute',
      bottom: '0',
      width: '100%',
    },
  },
  buttonDiv: {
    width: '100%',
  },
  button: {
    width: '100%',
    margin: '1px',
    color: 'white',
    backgroundColor: theme.palette.primary.A700,
    '&:hover': {
      backgroundColor: theme.palette.primary.A700,
      color: 'white',
    },
  },
});

const CreateChecklistForm = props => (
  <form
    id="create-list-form"
    onSubmit={props.handleSubmit}
  >
    <TextField
      required
      id="name"
      label="Title"
      value={props.title}
      onChange={props.handleTextFieldChange}
      helperText="Please give your checklist a title"
      fullWidth
      margin="normal"
      className={props.classes.textField}
    />
    {
      props.items.map((item, idx) => (
        <ChecklistItem
          key={item.id}
          item={item}
          idx={idx}
          handleItemTextChange={() => {}}
          handleItemDelete={() => {}}
        />
      ))
    }
    <div className={props.classes.buttonGroup}>
      <div className={props.classes.buttonDiv}>
        <Button
          id="add-item-button"
          className={props.classes.button}
          variant="flat"
          color="primary"
          onClick={props.handleAddItem}
        >
          Add Item
        </Button>
      </div>
      <div className={props.classes.buttonDiv}>
        <Button
          className={props.classes.button}
          variant="flat"
          color="primary"
        >
          Create List
        </Button>
      </div>
    </div>
  </form>
);

CreateChecklistForm.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleTextFieldChange: PropTypes.func.isRequired,
  handleAddItem: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
  })).isRequired,
};

CreateChecklistForm.defaultProps = {};

export default withStyles(styles)(CreateChecklistForm);
