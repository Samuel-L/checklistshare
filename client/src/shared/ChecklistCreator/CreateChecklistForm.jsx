import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from 'material-ui/styles';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';

import ChecklistItem from './ChecklistItem';

const styles = theme => ({
  textField: {
    margin: '5px',
  },
  buttonGroup: {
    [theme.breakpoints.up('lg')]: {
      position: 'fixed',
      right: '10%',
      top: '20%',
      width: '150px !important',
    },
  },
  buttonDiv: {
    width: '100%',
  },
  button: {
    width: '100%',
    margin: '1px',
    color: 'white',
    backgroundColor: theme.palette.primary.A700,
    '&:hover': {
      backgroundColor: theme.palette.primary.A700,
      color: 'white',
    },
  },
});

const CreateChecklistForm = props => (
  <form
    id="create-list-form"
    onSubmit={props.handleSubmit}
  >
    <TextField
      required
      id="title"
      label="Title"
      value={props.title}
      onChange={props.handleTitleChange}
      helperText="Please give your checklist a title"
      fullWidth
      margin="normal"
      className={props.classes.textField}
    />
    {
      props.items.map((item, idx) => (
        <ChecklistItem
          key={item.seq || item.id}
          item={item}
          idx={idx}
          handleItemTextChange={props.handleItemTextChange}
          handleItemDelete={props.handleDeleteItem}
        />
      ))
    }
    <div className={props.classes.buttonGroup}>
      <div className={props.classes.buttonDiv}>
        <Button
          id="add-item-button"
          className={props.classes.button}
          variant="flat"
          color="primary"
          onClick={props.handleAddItem}
        >
          Add Item
        </Button>
      </div>
      <div className={props.classes.buttonDiv}>
        <Button
          id="create-list-button"
          className={props.classes.button}
          variant="flat"
          color="primary"
          onClick={props.handleCreateList}
        >
          { props.editMode ? 'Update List' : 'Create List' }
        </Button>
      </div>
    </div>
  </form>
);

CreateChecklistForm.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleTitleChange: PropTypes.func.isRequired,
  handleItemTextChange: PropTypes.func.isRequired,
  handleAddItem: PropTypes.func.isRequired,
  handleDeleteItem: PropTypes.func.isRequired,
  handleCreateList: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
  })).isRequired,
  editMode: PropTypes.bool,
};

CreateChecklistForm.defaultProps = {
  editMode: false,
};

export default withStyles(styles)(CreateChecklistForm);
