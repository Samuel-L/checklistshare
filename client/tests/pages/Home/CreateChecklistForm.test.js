import React from 'react';
import renderer from 'react-test-renderer';

import CreateChecklistForm from '../../../src/pages/Home/CreateChecklistForm';

describe('pages/Home: CreateChecklistForm', () => {
  const classes = { textField: '', buttonGroup: '', buttonDiv: '', button: '' };
  const items = [{ id: 0, item: '' }];
  const tree = renderer.create(
    <CreateChecklistForm
      classes={classes}
      handleSubmit={jest.fn()}
      handleTextFieldChange={jest.fn()}
      handleAddItem={jest.fn()}
      handleDeleteItem={jest.fn()}
      title="Checklist title"
      items={items}
    />
  ).toJSON();
  
  it('should render', () => {
    expect(tree).toMatchSnapshot(); 
  });
});
