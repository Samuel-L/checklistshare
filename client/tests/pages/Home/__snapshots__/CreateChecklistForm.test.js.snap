// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Home: CreateChecklistForm should render 1`] = `
<form
  id="create-list-form"
  onSubmit={[MockFunction]}
>
  <div
    aria-describedby="name-helper-text"
    className="MuiFormControl-root-5 MuiFormControl-marginNormal-6 MuiFormControl-fullWidth-8 CreateChecklistForm-textField-1"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-14 MuiInputLabel-root-9 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12"
      data-shrink={true}
      htmlFor="name"
    >
      Title
      <span
        className="MuiFormLabel-asterisk-18"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-19 MuiInput-fullWidth-26 MuiInput-formControl-20 MuiInput-underline-23"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-27"
        defaultValue={undefined}
        disabled={false}
        id="name"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value="Checklist title"
      />
    </div>
    <p
      className="MuiFormHelperText-root-32"
      id="name-helper-text"
    >
      Please give your checklist a title
    </p>
  </div>
  <div
    className="ChecklistItem-root-36"
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-5 MuiFormControl-marginNormal-6 ChecklistItem-textField-37"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-14 MuiInputLabel-root-9 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
        data-shrink={false}
        htmlFor="item-0"
      >
        Item
        <span
          className="MuiFormLabel-asterisk-18"
        >
           *
        </span>
      </label>
      <div
        className="MuiInput-root-19 MuiInput-formControl-20 MuiInput-underline-23"
      >
        <input
          aria-invalid={false}
          aria-required={true}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-27"
          defaultValue={undefined}
          disabled={false}
          id="item-0"
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={true}
          rows={undefined}
          type="text"
          value={undefined}
        />
      </div>
    </div>
    <button
      aria-label="delete"
      className="MuiButtonBase-root-54 MuiButton-root-39 MuiButton-raised-44 ChecklistItem-icon-38"
      disabled={false}
      id="delete-item-button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-40"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-57"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-63"
      />
    </button>
  </div>
  <div
    className="CreateChecklistForm-buttonGroup-2"
  >
    <div
      className="CreateChecklistForm-buttonDiv-3"
    >
      <button
        className="MuiButtonBase-root-54 MuiButton-root-39 MuiButton-flatPrimary-41 CreateChecklistForm-button-4"
        disabled={false}
        id="add-item-button"
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-40"
        >
          Add Item
        </span>
        <span
          className="MuiTouchRipple-root-63"
        />
      </button>
    </div>
    <div
      className="CreateChecklistForm-buttonDiv-3"
    >
      <button
        className="MuiButtonBase-root-54 MuiButton-root-39 MuiButton-flatPrimary-41 CreateChecklistForm-button-4"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-40"
        >
          Create List
        </span>
        <span
          className="MuiTouchRipple-root-63"
        />
      </button>
    </div>
  </div>
</form>
`;
